libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +187,79 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam<MsFeatureFlagValue>[] {
+    const settings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    const featureFlags = new Array<MsFeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.dotnetFmSchemaKeyWord) {
+      const dotnetSchemaFeatureFlags = this.getDotnetFmSchemaFeatureFlags(config);
+
+      if (dotnetSchemaFeatureFlags) {
+        const featureManagementIndex = Constants.FeatureManagementKeyWords.indexOf(this.dotnetFmSchemaKeyWord);
+
+        for (const featureFlag in dotnetSchemaFeatureFlags) {
+          if (!this.validateFeatureName(featureFlag)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
+            );
+          }
+
+          const featureFlagValue = this.getFeatureFlagValueFromDotnetSchema(
+            featureFlag,
+            dotnetSchemaFeatureFlags[featureFlag],
+            Constants.EnabledForKeyWords[featureManagementIndex],
+            Constants.RequirementTypeKeyWords[featureManagementIndex]
+          );
+          
+          featureFlags.push(featureFlagValue);
+        }
       }
-      const prefix: string = options.prefix ?? "";
-      settings.push({
-        key: featureFlagPrefix + prefix + featureFlag,
+    }
+
+    if (this.foundMsFeatureManagement) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaFeatureFlags(config);
+
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              "Feature flag without id is found, id is a required property."
+            );
+          }
+
+          if (!this.validateFeatureName(featureFlag.id)) {
+            throw new ArgumentError(
+              `Feature flag id ${featureFlag.id} contains invalid character,'%' and ':' are not allowed.`
+            );
+          }
+
+          const featureFlagValue = this.getFeatureFlagValueFromMsFmSchema(featureFlag);
+
+          // Check if the featureFlag with the same id already exists
+          // Replace the existing flag with the later one, the later one always wins
+          const indexOfExistingFlag = featureFlags.findIndex(existingFeatureFlag => existingFeatureFlag.id === featureFlag.id);
```

Comment:
To confirm here on line 240 it means that if we find similar ff from the MsFmSchema then they take precedence ?

Reply:
Yes the MsFmSchema takes precedence